app.rb

```diff
@@ -93,6 +104,17 @@ def create_a_rental
     person = @people[personid]
     rental = Rental.new(date, book, person)
     @rental.push(rental)
+    store_all_rentals
+  end
+
+  def store_all_rentals
+    jsonarray = []
+    @rental.each do |item|
+      jsonarray.push({ date: item.date,
+                       book: { title: item.book.title, author: item.book.author },
+                       person: { id: item.person.id, name: item.person.name, age: item.person.age } })
+    end
+    File.write('rental.json', JSON.generate(jsonarray))
   end
 
   def list_all_rentals
```

Comment:
- [ ] **Required:** I observed that when a rental is created in a session of the app and the session is exited, the rentals is preserved. However, on reload, only the rental is not re-read into the app. Kindly check this out. [Check this for help.](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/store_data_in_files.md#storing-data-in-files)